lib/features/steps/runner_steps.rb

```diff
@@ -150,6 +163,20 @@
   assert(match, "No output lines from #{current_shell.stdout_lines} matched #{expected_line}")
 end
 
+# Wait for a string to appear in the stdout logs
+#
+# @step_input expected_line [String] The string present in stdout logs
+Then('I wait for the shell to output {string} to stdout') do |expected_line|
+  wait = Maze::Wait.new(timeout: MazeRunner.config.receive_requests_wait)
+  current_shell = Runner.get_interactive_session
+
+  success = wait.until do
+    current_shell.stdout_lines.any? { |line| line == expected_line }
```

Comment:
I think it is fine for now, given there is a timeout, but we might need to introduce a separate step one day to give a specific timeout for longer running processes.

Reply:
...something along the lines of `I wait for up to {int} seconds for the shell...`.

Reply:
Yeah, if we need to build a whole RN app that'll probably be necessary. I might leave it out for now as it should be super quick to add and we might not end up doing that in the end